checkov/common/output/report.py

```diff
@@ -72,7 +73,34 @@ def get_dict(self, is_quiet=False) -> dict:
                 "summary": self.get_summary()
             }
 
-    def get_exit_code(self, soft_fail: bool) -> int:
+    def get_exit_code(self, soft_fail: bool, soft_fail_on: Union[list, None] = None,
+                      hard_fail_on: Union[list, None] = None) -> int:
+        """
+        Returns the appropriate exit code depending on the flags that are passed in.
+
+        :param soft_fail: If true, exit code is always 0. (default is false)
+        :param soft_fail_on: A list of checks that will return exit code 0 if they fail. Other failing checks will
+        result exit code 1.
+        :param hard_fail_on: A list of checks that will return exit code 1 if they fail. Other failing checks will
+        result exit code 0.
+        :return: Exit code 0 or 1.
+        """
+        if soft_fail_on:
+            soft_fail_on = convert_csv_string_arg_to_list(soft_fail_on)
+            if all(check_id in soft_fail_on for check_id in
+                   (failed_check.check_id for failed_check in self.failed_checks)):
+                # List of "failed checks" is a subset of the "soft fail on" list.
+                return 0
+            else:
+                return 1
+        if hard_fail_on:
+            hard_fail_on = convert_csv_string_arg_to_list(hard_fail_on)
+            if any(check_id in hard_fail_on for check_id in
+                   (failed_check.check_id for failed_check in self.failed_checks)):
+                # Any check from the list of "failed checks" is in the list of "hard fail on checks".
+                return 1
+            else:
+                return 0
         if soft_fail:
```

Comment:
You're right!

Reply:
Since these parameters are mutually exclusive (as introduced in `main.py`) - I'm not sure we should even go down the rabbithole of order and priority. As long as they are mutually exclusive, whatever you define will be the way we implement, so we're good